packages/framework-provider-aws/src/library/events-searcher-adapter.ts

```diff
@@ -0,0 +1,240 @@
+  logger.debug('Initiating an events search. Filters: ', filters)
+  const timeFilterQuery = buildSearchEventsTimeQuery(filters.from, filters.to)
+  const eventEnvelopes = await executeSearch(dynamoDB, config, logger, filters, timeFilterQuery)
+
+  logger.debug('Events search result: ', eventEnvelopes)
+  return convertToSearchResult(eventEnvelopes)
+}
+
+interface TimeQueryData {
+  expression: string
+  attributeValues: Record<string, string>
+}
+
+function buildSearchEventsTimeQuery(from?: string, to?: string): TimeQueryData {
+  let timeQueryData: TimeQueryData = {
+    expression: '',
+    attributeValues: {},
+  }
+  if (from && to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} BETWEEN :fromTime AND :toTime`,
+      attributeValues: {
+        ':fromTime': from,
+        ':toTime': to,
+      },
+    }
+  } else if (from) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} >= :fromTime`,
+      attributeValues: { ':fromTime': from },
+    }
+  } else if (to) {
+    timeQueryData = {
+      expression: ` AND ${eventsStoreAttributes.sortKey} <= :toTime`,
+      attributeValues: { ':toTime': to },
+    }
+  }
+  return timeQueryData
+}
+
+async function executeSearch(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  filters: EventFilter,
+  timeFilterQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  if ('entity' in filters) {
+    if (filters.entityID) {
+      return await searchEventsByEntityAndID(
+        dynamoDB,
+        config,
+        logger,
+        filters.entity,
+        filters.entityID,
+        timeFilterQuery
+      )
+    } else {
+      return await searchEventsByEntity(dynamoDB, config, logger, filters.entity, timeFilterQuery)
+    }
+  } else if ('type' in filters) {
+    return await searchEventsByType(dynamoDB, config, logger, filters.type, timeFilterQuery)
+  } else {
+    throw new Error('Invalid search event query. It is neither an search by "entity" nor a search by "type"')
+  }
+}
+
+async function searchEventsByEntityAndID(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entity: string,
+  entityID: UUID,
+  timeQuery: TimeQueryData
+): Promise<Array<EventEnvelope>> {
+  // TODO: Manage pagination
+  const params: DocumentClient.QueryInput = {
+    TableName: config.resourceNames.eventsStore,
+    ConsistentRead: true,
+    ScanIndexForward: false, // Descending order (newer timestamps first)
+    KeyConditionExpression: `${eventsStoreAttributes.partitionKey} = :partitionKey${timeQuery.expression}`,
+    ExpressionAttributeValues: {
+      ...timeQuery.attributeValues,
+      ':partitionKey': partitionKeyForEvent(entity, entityID),
+    },
+  }
+
+  logger.debug('Searching events by entity and entity ID. Query params: ', params)
+  const result = await dynamoDB.query(params).promise()
+  return (result.Items as Array<EventEnvelope>) ?? []
```

Comment:
To add pagination easily:
```suggestion
  const resultItems: Array<EventEnvelope> = []
  let result: PromiseResult<somethingfromaws, AWSError>  // Check this type
  // For pagination, just wrap in a do..while
  do {
    result = await dynamoDB.query(params).promise()
    resultItems.concat(result.Items as Array<EventEnvelope> ?? [])
  } while(result.LastEvaluatedKey)
  return resultItems
```

Reply:
Two things:
- How is this code dealing with pagination? I don't get it
- The kind of pagination that this method needs is another one: the ability to specify a pointer (`lastEvaluatedKey`) and page size to get a specific page. It is related to GraphQL.